### PR TITLE
Online book store spring boot security v2

### PR DESCRIPTION
**Pull request by sshestaka**

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstore/controller/AuthController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@RequiredArgsConstructor
+@RequestMapping("/api/auth")
+@Tag(name = "Authentication", description = "Endpoints for login or register a new user")
+public class AuthController {
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    @PostMapping("/login")
+    @Operation(summary = "Login page", description = "Login an existed user")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto requestDto) {
```
Comment (Ivan95kos) [inline src/main/java/mate/academy/onlinebookstore/controller/AuthController.java]:
```suggestion
    public UserLoginResponseDto login(@RequestBody @Valid UserLoginRequestDto requestDto) {
```

File: src/main/java/mate/academy/onlinebookstore/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
+        }
```
Comment (Ivan95kos) [inline src/main/java/mate/academy/onlinebookstore/security/JwtAuthenticationFilter.java]: move the Strings and number to constants

File: src/main/java/mate/academy/onlinebookstore/service/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
+    private final RoleRepository roleRepository;
+
+    private void updatedUser(User updatedUser, UpdateUserRequestDto updateUserRequestDto) {
```
Comment (Ivan95kos) [inline src/main/java/mate/academy/onlinebookstore/service/UserServiceImpl.java]: All private methods should be below public methods

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
Good job overall!

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**